lib/elastic_apm.rb

```diff
@@ -381,5 +391,29 @@ def add_filter(key, callback = nil, &block)
 
       agent&.add_filter(key, block || callback)
     end
+
+    def set_transaction_outcome(result: nil, http_status_code: nil)
```

Comment:
I'm not sure why we want a public(?) API for this?

As far as I can see, we already have direct access to either `transaction` or `span` where the outcome is set, so couldn't it instead just be set as a simple prop?